api-helpers/articles.ts

```diff
@@ -1,135 +1,72 @@
 
-import { isIsoDate } from '../utils/date-utils';
+import { LIST_ARTICLES_PER_PAGE, TILES_ARTICLES_PER_BLOG, TILES_BLOGS_PER_PAGE } from '../contants';
 
-import { cursorToText, textToCursor } from './cursor-encoding';
 import { HTTPNotFound } from './errors';
 
-export const TILES_BLOGS_PER_PAGE = 4;
-export const TILES_ARTICLES_PER_BLOG = 5;
-export const LIST_ARTICLES_PER_PAGE = 20;
-
-function last<T extends readonly R[], R>(arr: readonly [R, ...T]): R;
-function last<T>(arr: readonly T[]): T | undefined;
-function last<T>(arr: readonly T[]): T | undefined {
-  return arr[arr.length - 1];
+function pageValidGuard(page: string | undefined): void {
+  if (page === undefined) throw new HTTPNotFound('Page must be a number value');
+  if (isNaN(+page)) throw new HTTPNotFound('Page must be a number value');
+  return;
 }
 
-export const getArticlesForGrid = async (prisma: PrismaClient, cursor?: string) => {
-  const date = cursor && cursorToText(cursor);
-
-  if (date && !isIsoDate(date)) {
-    throw new HTTPNotFound();
-  }
-
-  const where = date
-    ? {
-        lastArticlePublishedAt: {
-          lt: date,
-          not: null,
-        },
-      }
-    : {};
-
+export const getArticlesForGrid = async (prisma: PrismaClient, page?: string) => {
+  pageValidGuard(page);
   const blogs = await prisma.blog.findMany({
-    where: { ...where, isPublic: true },
+    where: { isPublic: true },
+    skip: page ? +page * TILES_BLOGS_PER_PAGE + 1 : 0,
     take: TILES_BLOGS_PER_PAGE,
     orderBy: {
-      lastArticlePublishedAt: 'desc',
+      lastArticlePublishedAt: 'asc',
     },
     include: {
       articles: {
         take: TILES_ARTICLES_PER_BLOG,
         orderBy: {
-          publishedAt: 'desc',
+          publishedAt: 'asc',
         },
       },
     },
   });
+  if (blogs.length === 0) throw new HTTPNotFound();
 
-  const lastBlog = last(blogs);
   return {
-    data: blogs,
-    nextCursor:
-      lastBlog?.lastArticlePublishedAt &&
-      textToCursor(lastBlog?.lastArticlePublishedAt.toISOString()),
+    data: blogs.reverse(),
   };
 };
 
-export const getArticlesPaginationForGrid = async (prisma: PrismaClient) => {
-  const blogs = (await prisma.blog.findMany({
-    where: { isPublic: true, lastArticlePublishedAt: { not: null } },
-    orderBy: {
-      lastArticlePublishedAt: 'desc',
-    },
-    select: {
-      lastArticlePublishedAt: true,
-    },
-  })) as ReadonlyArray<{ readonly lastArticlePublishedAt: Date }>;
-
-  const cursors = blogs.flatMap((blog, index) => {
-    if (index % TILES_BLOGS_PER_PAGE === TILES_BLOGS_PER_PAGE - 1) {
-      return [textToCursor(blog.lastArticlePublishedAt.toISOString())];
-    }
-    return [];
+export const getLastArticlePage = async (prisma: PrismaClient) => {
+  const articlesCount = await prisma.article.count({
+    where: { blog: { isPublic: true } },
   });
-  return cursors;
+  return Math.floor(articlesCount / LIST_ARTICLES_PER_PAGE);
```

Comment:
Nie powinno być przypadkiem zaokrąglone w górę?
```suggestion
  return Math.ceil(articlesCount / LIST_ARTICLES_PER_PAGE);
```

Reply:
tak. przez to właśnie ostatnia strona zostaje nie pełna.